indexing-hadoop/src/main/java/io/druid/indexer/IndexGeneratorJob.java

```diff
@@ -374,9 +378,9 @@ protected void reduce(
         Set<String> allDimensionNames = Sets.newHashSet();
         final ProgressIndicator progressIndicator = makeProgressIndicator(context);
 
-        for (final Text value : values) {
+        for (final Writable value : values) {
           context.progress();
-          final InputRow inputRow = index.formatRow(parser.parse(value.toString()));
+          final InputRow inputRow = index.formatRow(HadoopDruidIndexerMapper.parseInputRow(value, parser));
```

Comment:
@nishantmonu51  only reason I made it public static is that IndexGeneratorJob.IndexGeneratorReducer needs it as well. however, i still couldn't convince myself that it should go to a new class.

should we move it to io.druid.indexer.Utils class or create a brand new class? I am actually fine keeping it in HadoopDruidIndexerMapper though.